### DISCUSSION TITLE
Extconf Failed

### DISCUSSION THREAD

**Discussion by ZuluPilgrim** [pglombardo/PasswordPusher#928] [Q&A]
Where do I start looking to work out why the following happened? I tried googling but am not sure what I am googling for. It appears to be a Ruby issue but I did install in on my Ubuntu build.

![image](https://user-images.githubusercontent.com/111164245/225540353-484a9cb2-cb3d-4074-8013-c3390265ba31.png)

**Comment by pglombardo** [comment]
Hi @ZuluPilgrim  - That gem has a c extension that needs to be compiled - but is failing.  Most likely you need to install some development libraries or tools to compile.  To find out, could you post the contents of that `mkmf.log` file?  That should tell us where the error was.

**Reply by ZuluPilgrim** [reply]
Unfortunately I had already started on a new build and am now getting a different error which I will ask about in a new thread.

**Comment by pglombardo** [comment]
@ZuluPilgrim I will close this discussion since it seems you've moved onto Docker (which is the right move IMO).  The source code development environment can be a bit tricky.
